doc/ExpertOptions.md

```diff
@@ -983,11 +984,88 @@ into stage inputs. The `add` stage will have compulsory integer inputs
 
 For an in depth discussion, please see [Missing Call Arguments](MissingCallArguments.md).
 
-# Docker
+# Manifests
+
+In extreme cases, running compiled WDL workflows can fail due to DNAnexus platform limits on the total size of the input and output JSON documents of a job. An example is a task with many inputs/outputs that is called in scatter over a large collection. In such a case, you can enable manifest support at compile time with the `-useManifests` option. This option causes each generated applet or workflow to accept inputs as a manifest, and to produce outputs as a manifest.
+
+A manifest is a JSON document that contains all the inputs/outputs that would otherwise be passed directly to/from the applet. A manifest can be specified in one of two ways: via a JSON input, or via a File input (where the file must exist on the platform).
+
+## Manifest JSON
+
+When manifest support is enabled, each applet has an `input_mainfest___` input field of type `hash`, which means that it accepts a JSON document as a string. For example, given the following workflow:
+
+```wdl
+workflow test {
+  input {
+    String s
+    File f
+  }
+  ...
+  output {
+    Int i
+    Pair[String, File] p
+  }
+}
+```
+
+You would write the following manifest:
+
+`mymanifest.json`
+```json
+{
+  "test.input_manifest___": {
+    "s": "hello",
+    "f": "dx://file-xxx"
+  }
+}
+```
+
+When you compile the workflow, provide the manifest using the `-inputs` option, and it will be translated to:
+
+`mymanifest.dx.json`
+```json
+{
+  "input_manifest___": {
+    "s": "hello",
+    "f": {
+      "$dnanexus_link": "file-xxx"
+    }
+  }
+}
+```
 
-As of release 0.80, we moved to using docker, instead of
-[dx-docker](https://documentation.dnanexus.com/developer/apps/dependency-management/using-docker-images). `dx-docker`
-is deprecated, although you can still use it with the `--useDxDocker` command line flag.
+Finally, run your workflow using the translated input file:
+
+`dx run workflow-yyy -f mymanifest.dx.json`
+
+## Manifest file
+
+Manifest files are less convenient to use as applet/workflow inputs because they must be uploaded to the platform. However, when manifest support is enabled, applet/workflow outputs are in the form of manifest files, so it is useful to understand the format.
+
+Given the above workflow, the manifest output would be:
+
+```json
+{
+  "id": "test",
+  "values": {
+    "i": 1,
+    "p": {
+      "left": "hello",
+      "right": "dx://file-xxx"
+    }
+  }
+}
+```
+
+The `id` field is optional but will always be populated in the output manfiests. The manifest may contain additional fields (`types` and `definitions`) that are only for internal use and can be ignored.
```

Comment:
It'd be good to mention where will the output manifest be uploaded and how (if) they can be used?

Reply:
done